x-pack/plugins/watcher/server/models/visualize_options/visualize_options.js

```diff
@@ -9,7 +9,8 @@ export class VisualizeOptions {
   constructor(props) {
     this.rangeFrom = props.rangeFrom;
     this.rangeTo = props.rangeTo;
-    this.interval = props.interval;
+    this.interval = props.interval; // Only used in 7.x, will be undefined in 8.x
+    this.fixed_interval = props.fixed_interval; // Used from 8.x
```

Comment:
Is it necessary to differentiate between `fixed_interval` and `interval` at this level? These are ES concepts, but during the definition of the visualization in the UI and the transmission to the server, it seems like we only have a single concept of "interval". Can we get away with waiting to differentiate between which one is being used until we get to lines 112 and 113 of `build_visualize_query.js`?

Reply:
Good idea. I didn't want to make kibana version check server side to not rely on the constant but I don't see any problem with relying on the constant for the kibana version.